### PR TITLE
build: Micronaut Framework 4.0.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Comment by sdelamo** [comment]
@wetted can you merge `master` against this branch and get this PR ready?. Thanks!

File: buildSrc/src/main/groovy/io.micronaut.build.internal.problem-json-base.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        maven { setUrl("https://s01.oss.sonatype.org/content/repositories/snapshots/") }
+    }
+}
+
+configurations.all {
+    resolutionStrategy.dependencySubstitution {
+        substitute(module("org.codehaus.groovy:groovy"))
+                .using(module("org.apache.groovy:groovy:${libs.versions.groovy}"))
+    }
+}
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.problem-json-base.gradle]: We shouldn't need this any more 🤔

Comment (timyates) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.problem-json-base.gradle]: Pushed a fix https://github.com/micronaut-projects/micronaut-problem-json/pull/212/commits/78166cb5acb029afb2b817e696004448eb6ac93d

**Review by timyates — APPROVED** [review]

File: problem-json/build.gradle.kts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 dependencies {
-    annotationProcessor(mn.micronaut.serde.processor)
+    annotationProcessor(mnSerde.micronaut.serde.processor)
     api(libs.managed.zalando.problem)
-    implementation(mn.micronaut.serde.api)
+    implementation(mnSerde.micronaut.serde.api)
     implementation(mn.micronaut.validation)
     implementation(mn.micronaut.http.server)
+    implementation(mn.micronaut.jackson.databind)
```
Comment (sdelamo) [inline problem-json/build.gradle.kts:11]: why is this necessary? are not we using serde?

Comment (wetted) [reply problem-json/build.gradle.kts:11]: I fixed the dependencies, but now I get a test failure in `test-suite`. Investigating.

Comment (timyates) [reply problem-json/build.gradle.kts:11]: Just had a quick look...  When we switch to Serde, we do not get a `ConstraintViolationException` anymore (ie: in `ConstraintViolationProblemSpec`), we get an error:

```
io.micronaut.core.convert.exceptions.ConversionErrorException: Failed to convert argument [contact] for value [null] due to: Unable to deserialize type [io.micronaut.problem.ConstraintViolationProblemSpec$ContactCreateForm]. Required property [String name] is not present in supplied data
```

This means `ProblemConstraintViolationExceptionHandler` doesn't get invoked

Needs more thought... 🤔

Comment (wetted) [reply problem-json/build.gradle.kts]: @timyates yeah, we have three tests failing now, and I think at least two are the problem you mention.

Comment (sdelamo) [reply problem-json/build.gradle.kts]: Fixed it here: https://github.com/micronaut-projects/micronaut-problem-json/pull/212/commits/3de7e529b7191ea57eadaa301bdd4dbd0e4bdf8e
